client/src/js/app.js

```diff
@@ -1253,43 +1263,59 @@ function attachSignaturePad() {
 
   window.onresize = resizeCanvas;
   resizeCanvas();
-
-  // signaturePadCanvas = document.querySelector(signaturePadSelector);
-  // console.log(signaturePadCanvas);
-  // if (signaturePadCanvas === null) {
-  //   console.err('Signature widget failure');
-  //   return;
-  // }
-  // signaturePadElement = document.querySelector('#signatureCanvas');
-  // signaturePad = new SignaturePad(signaturePadCanvas, {
-    // 'onEnd': function () {
-    //   var event = makeCustomEvent("signatureupdate", {
-    //     file: signaturePad.toDataURL("image/svg+xml")
-    //   });
-    //
-    //   signaturePadElement.dispatchEvent(event);
-    // }
-  // });
-  // var signature_data = signaturePadElement.getAttribute('data-signature');
-  // if (signature_data.length > 0) {
-  //   signaturePadElement.fromDataURL(signature_data);
-  // }
-  // window.addEventListener("resize", signaturePadResizeCanvas);
-  // signaturePadResizeCanvas();
 }
 
+function storeSignatureFromPad() {
+  (async () => {
+    const uploadCache = 'photos-upload';
+    const cache = await caches.open(uploadCache);
+
+    signaturePad.canvas.toBlob(async function(blob) {
+      const formData = new FormData();
+      const imageName = 'signature-' + getRandom8Digits() + '.png';
+      formData.set('file', blob, imageName);
+
+      const url = "cache-upload/images/" + Date.now();
+
+      try {
+        var response = await fetch(url, {
+          method: 'POST',
+          body: formData,
+          // This prevents attaching cookies to request, to prevent
+          // sending authentication cookie, as our desired
+          // authentication method is token.
+          credentials: 'omit'
+        });
 
+        if (response.ok) {
+         var json = await response.json();
+         reportSignaturePadResult(json.url);
+        }
+        else {
+          // If something goes wrong while storing signature in cache,
+          // currently we do nothing.
+          // This situation is very rare, and if it does happen, user
+          // will most likely repeat the action.
```

Comment:
Why not `console.warn` the event?

Reply:
Because the APP is used on tablet / mobile phone, so noone would see console.log anyway.